File: Android/app/src/main/java/com/example/smartfashion/model/CommunityPost.kt
package com.example.smartfashion.model

import com.google.gson.annotations.SerializedName
import java.util.Date

data class CommunityPost(
    @SerializedName("post_id")
    val postId: Int? = null,

    @SerializedName("user_id")
    val userId: Int? = null,

    @SerializedName("outfit_id")
    val outfitId: Int? = null,

    @SerializedName("image_url")
    val imageUrl: String,

    @SerializedName("description")
    val description: String? = null,

    @SerializedName("likes_count")
    val likesCount: Int = 0,

    @SerializedName("height_ratio")
    val heightRatio: Float,

    // ------

    @SerializedName("author_name")
    val authorName: String? = null,

    @SerializedName("author_avatar")
    val authorAvatar: String? = null,

    @SerializedName("tags")
    val tags: List<String>? = null,

    @SerializedName("is_liked")
    val isLiked: Boolean = false,

    @SerializedName("created_at")
    val createdAt: Date? = null,

    @SerializedName("updatedAt")
    val updatedAt: Date? = null
)

File: Server/models/CommunityPost.js
const mongoose = require('mongoose');
const Counter = require('./Counter');

const communityPostSchema = new mongoose.Schema({
  post_id: {
    type: Number,
    unique: true
  },
  user_id: {
    type: Number,
    required: true,
    ref: 'User',
    index: true 
  },
  outfit_id: {
    type: Number,
    required: true,
    ref: 'Outfit',
    index: true
  },
  image_url: {
    type: String,
    required: true
  },
  description: {
    type: String,
    default: null
  },
  likes_count: {
    type: Number,
    default: 0
  },
  height_ratio: {
    type: Number,
    required: true
  },
  created_at: {
    type: Date,
    default: Date.now,
    index: -1
  }
}, {
  timestamps: true
});

communityPostSchema.pre('save', async function() {
  const doc = this;

  if (!doc.isNew) {
    return;
  }

  const counter = await Counter.findByIdAndUpdate(
    { _id: 'post_id' }, 
    { $inc: { seq: 1 } },
    { new: true, upsert: true }
  );

  doc.post_id = counter.seq;
});

module.exports = mongoose.model('CommunityPost', communityPostSchema, 'community_posts');